dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/PreCheckSecurityOwnershipValidationHook.java

```diff
@@ -226,8 +271,16 @@ private String getTeiUidFromEvent( TrackerImportValidationContext context, Event
 
         if ( programInstance == null )
         {
-            Enrollment enrollment = context.getBundle().getEnrollment( event.getEnrollment() ).get();
-            return enrollment.getTrackedEntity();
+            Optional<Enrollment> optionalEnrollment = context.getBundle().getEnrollment( event.getEnrollment() );
+            if ( optionalEnrollment.isPresent() )
+            {
+                Enrollment enrollment = optionalEnrollment.get();
+                return enrollment.getTrackedEntity();
+            }
+            else
+            {
+                return null;
+            }
```

Comment:
Maybe here we can use something more "functional" to make the code more coincise?
```suggestion
            return context.getBundle()
                    .getEnrollment( event.getEnrollment() )
                    .map( Enrollment::getTrackedEntity )
                    .orElse(null );
```